test/v1/controllers/AmendForeignPropertyControllerSpec.scala

```diff
@@ -0,0 +1,257 @@
+      "parser errors occur" should {
+        def errorsFromParserTester(error: MtdError, expectedStatus: Int): Unit = {
+          s"a ${error.code} error is returned from the parser" in new Test {
+
+            MockAmendForeignPropertyRequestParser
+              .parseRequest(rawData)
+              .returns(Left(ErrorWrapper(Some(correlationId), error, None)))
+
+            val result: Future[Result] = controller.handleRequest(nino, businessId, submissionId)(fakePostRequest(requestJson))
+
+            status(result) shouldBe expectedStatus
+            contentAsJson(result) shouldBe Json.toJson(error)
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+          }
+        }
+
+        val input = Seq(
+          (BadRequestError, BAD_REQUEST),
+          (NinoFormatError, BAD_REQUEST),
+          (BusinessIdFormatError, BAD_REQUEST),
+          (SubmissionIdFormatError, BAD_REQUEST),
+          (CountryCodeFormatError.copy(paths = Some(Seq(
+            "foreignProperty/0/countryCode"))), BAD_REQUEST),
+          (ValueFormatError.copy(paths = Some(Seq(
+            "foreignFhlEea/income/rentAmount",
+            "foreignFhlEea/expenditure/repairsAndMaintenance",
+            "foreignFhlEea/expenditure/professionalFees",
+            "foreignFhlEea/expenditure/other",
+            "foreignProperty/income/rentIncome/rentAmount",
+            "foreignProperty/expenditure/professionalFees",
+            "foreignProperty/expenditure/other"))), BAD_REQUEST),
+          (RuleIncorrectOrEmptyBodyError, BAD_REQUEST),
+          (RuleBothExpensesSuppliedError, BAD_REQUEST),
+          (RuleCountryCodeError.copy(paths = Some(Seq(
+            "foreignProperty/0/countryCode"))), BAD_REQUEST)
+        )
+
+        input.foreach(args => (errorsFromParserTester _).tupled(args))
+      }
+
+      "service errors occur" should {
+        def serviceErrors(mtdError: MtdError, expectedStatus: Int): Unit = {
+          s"a $mtdError error is returned from the service" in new Test {
+
+            MockAmendForeignPropertyRequestParser
+              .parseRequest(rawData)
+              .returns(Right(requestData))
+
+            MockAmendForeignPropertyService
+              .amend(requestData)
+              .returns(Future.successful(Left(ErrorWrapper(Some(correlationId), mtdError))))
+
+            val result: Future[Result] = controller.handleRequest(nino, businessId, submissionId)(fakePostRequest(requestJson))
+
+            status(result) shouldBe expectedStatus
+            contentAsJson(result) shouldBe Json.toJson(mtdError)
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+          }
+        }
+
+        val input = Seq(
+          (NinoFormatError, BAD_REQUEST),
+          (BusinessIdFormatError, BAD_REQUEST),
+          (SubmissionIdNotFoundError, NOT_FOUND),
+          (NotFoundError, NOT_FOUND),
+          (DownstreamError, INTERNAL_SERVER_ERROR)
+        )
```

Comment:
Spoke to francis and asked him if its assumed that DES returns an `INVALID_SUBMISSION_ID` or not since the retrieve has it as also backend and the amend has it as frontend only.

Its being treated also as a backend error so I believe it should be added to here as well as the other related places like the service and ISpec.